src/Utils/TimeDependentGenerator.cpp

```diff
@@ -313,6 +281,39 @@ MINT::counted_ptr<IDalitzEvent> TimeDependentGenerator::generate_event() const {
   int tag = generate_tag() ;
   double decaytime = generate_decay_time(tag) ;
   double smeareddecaytime = -999. ;
+
+  smeareddecaytime = decaytime + m_rndm->Gaus(0, m_resWidth) ;
+
+  if ( (m_h_efficiency != NULL) && (m_addExpEffects) ){
+    int i = 0 ;
+    float r_num = 0, efficiency = 0 ;
+    int maxiter = 100000 ;
+    while(true){
+      if( smeareddecaytime < m_efficiencyFit.GetXmax() ){
```

Comment:
@NiallMcHugh - Actually, here you should check if `smeareddecaytime` is within the range of `m_efficiencyFit`, so you don't assume that `m_efficiencyFit` goes down to 0. If `smeareddecaytime < m_efficiencyFit.GetXmin()` use the efficiency as evaluated at `m_efficiencyFit.GetXmin()`. Similarly, if `smeareddecaytime > m_efficiencyFit.GetXmax()`, use the efficiency as evaluated at `m_efficiencyFit.GetXmax()`, so you don't assume that the maximum efficiency is exactly 1.

Reply:
@MannyMoo I've added a new commit where this should be fixed.

Reply:
Yep, looks good!